src/services/toil-accruals.service.ts

```diff
@@ -0,0 +1,60 @@
+import { AxiosResponse } from 'axios';
+import { Options, RequirementsOf, Service } from './service';
+import { ToilAccrualsQueryParams } from '../interfaces/query-params/toil-accruals-query-params.interface';
+import { ToilAccrual } from '../models/toil-accrual.model';
+import { ApiToilAccrual } from '../interfaces/toil-accrual.interface';
+
+type RequiredProps = 'duration_hours' | 'date' | 'leave_year' | 'user_id';
+
+export class ToilAccrualsService extends Service {
+  private apiPath = '/toil_accruals';
+
+  create(data: RequirementsOf<ApiToilAccrual, RequiredProps>): Promise<ToilAccrual>;
+  create(
+    data: RequirementsOf<ApiToilAccrual, RequiredProps>,
+    options: { rawResponse: true } & Options
+  ): Promise<AxiosResponse<ApiToilAccrual, any>>;
+  create(data: RequirementsOf<ApiToilAccrual, RequiredProps>, options: Options): Promise<ToilAccrual>;
+  create(data: RequirementsOf<ApiToilAccrual, RequiredProps>, options?: Options) {
+    return super
+      .fetch<ToilAccrual>({ url: this.apiPath, data, method: 'POST' })
+      .then((res) => Promise.resolve(options?.rawResponse ? res : new ToilAccrual(res.data)));
+  }
+
+  get(id: number): Promise<ToilAccrual>;
+  get(id: number, options: { rawResponse: true } & Options): Promise<AxiosResponse<ToilAccrual, any>>;
+  get(id: number, options: Options): Promise<ToilAccrual>;
+  get(id: number, options?: Options) {
+    return super
+      .fetch<ToilAccrual>({ url: `${this.apiPath}/${id}` }, options)
+      .then((res) => Promise.resolve(options?.rawResponse ? res : new ToilAccrual(res.data)));
+  }
+
+  async *list(query: ToilAccrualsQueryParams, options?: Options) {
+    for await (const res of super.iterator<ApiToilAccrual>({ url: this.apiPath, params: query }, options)) {
+      yield new ToilAccrual(res);
+    }
+  }
+
+  listAll(query: ToilAccrualsQueryParams, options?: Options): Promise<ToilAccrual[]>;
+  async listAll(query: ToilAccrualsQueryParams, options?: Options) {
```

Comment:
Unecessary overload?
```suggestion
  async listAll(query: ToilAccrualsQueryParams, options?: Options): Promise<ToilAccrual[]> {
```

Reply:
good shout have removed, these do match those from the other services - any idea why they might have been needed?

Reply:
Oh! Not sure doesn't look like they are needed. Guessing it was a half complete overload for the `rawResponse` maybe?  🤷‍♂️